src/context.ts

```diff
@@ -14,27 +14,24 @@ import {
   useStore,
 } from 'zustand'
 
-export type UseContextStore<T extends State> = {
-  (): T
-  <U>(selector: StateSelector<T, U>, equalityFn?: EqualityChecker<U>): U
-}
+export type UseContextStore<S> = <U = ExtractState<S>>(
```

Comment:
Since, that type is changed. We can rename the new one and deprecate the old one in #715?

Reply:
I'm not sure about renaming the new one (as the current name is good), what we can do is we can handle both cases...

```
type UseContextStore<T extends State | Store<State>> = 
  <U = T extends Store<State> ? ExtractState<T> : T>(
    selector?: (state: ExtractState<S>) => U,
    equals?: (a: U, b: U) => boolean
  ) => U
```

A little hacky but I think it'll work. Wdyt?

Reply:
Made this change in #772, will revert if we're doing something else

Reply:
I will look into it. We want to mark `@deprecated` to the old usage.

Reply:
We agree to stop exporting this type.

Reply:
First I mistakenly removed it entirely but now I'm keeping it just won't export it (so that the tooltip is a better)